x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/core/security/transport/ProfileConfigurations.java

```diff
@@ -8,25 +8,61 @@
 
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.ssl.SslConfiguration;
-import org.elasticsearch.common.util.Maps;
 import org.elasticsearch.transport.TransportSettings;
 import org.elasticsearch.xpack.core.XPackSettings;
 import org.elasticsearch.xpack.core.ssl.SSLService;
 
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 import static org.elasticsearch.transport.RemoteClusterPortSettings.REMOTE_CLUSTER_PORT_ENABLED;
 import static org.elasticsearch.transport.RemoteClusterPortSettings.REMOTE_CLUSTER_PROFILE;
+import static org.elasticsearch.xpack.core.XPackSettings.REMOTE_CLUSTER_SSL_ENABLED;
 import static org.elasticsearch.xpack.core.security.SecurityField.setting;
 
 public final class ProfileConfigurations {
```

Comment:
Can you add some java doc to this class ? Specifically calling out the difference between transport internal, transport for legacy remote cluster connections and now the new dedicated remote cluster port that is backed but not configurable via the named profile.

Reply:
That's a good suggestion. I added [javadoc](https://github.com/elastic/elasticsearch/pull/93334/commits/cd4cca93464202b34d5428cde2ee312175b22eb1#diff-4464d67e0e26bda7adba76f549b07d1e8cc1ddd122fa37b4edd40558b351884f) for both the class and the method. Please let me know whether it reads correct for you. Thanks!